### PR TITLE
Refactoring/ferbemar/72 refactor mapping design

### PR DESCRIPTION
**Pull request by Ra1nerZufall**
Delete Tasks on PlanProject and TogglProject side.
Both corresponding projects have a dictionary with taskId and name.
all entries and repetitionEntries have the id of their task or -1 if non is assigned

### CODE REVIEW

File: AcademicTimePlanner/DataMapping/Plan/PlanProject.cs [hunk 1]
```diff
@@ -35,25 +44,89 @@ public PlanProject(long togglProjectId, string name)
...
+            if (_repetitionEntries == null)
+                return (from planEntry in _planEntries.FindAll(planEntry => planEntry.StartDate >= startDate && planEntry.EndDate <= endDate) select planEntry.Duration).Sum();
+
+            return (from planEntry in _planEntries.FindAll(planEntry => planEntry.StartDate >= startDate && planEntry.EndDate <= endDate) select planEntry.Duration).Sum() +
+                    (from repetitionEntry in _repetitionEntries select repetitionEntry.GetDurationInTimeRange(startDate, endDate)).Sum();
+        }
+
+        public List<PlanEntry> GetAllPlanEntriesList()
+        {
+            var planEntries = new List<PlanEntry>();
```
Comment (DuplosFidibuss) [inline AcademicTimePlanner/DataMapping/Plan/PlanProject.cs:110]: Does this need to be public?

File: AcademicTimePlanner/DataMapping/Plan/PlanProject.cs [hunk 2]
```diff
@@ -35,25 +44,89 @@ public PlanProject(long togglProjectId, string name)
...
+        public void RemoveRepetitionEntry(PlanEntryRepetition planEntryRepetition)
         {
-            _taskList.Remove(planTask);
+            _repetitionEntries.Remove(planEntryRepetition);
+        }
+
+        public double GetDurationInTimeRange(DateTime startDate, DateTime endDate)
+        {
+            if (_planEntries == null && _repetitionEntries == null)
+                return 0;
```
Comment (DuplosFidibuss) [inline AcademicTimePlanner/DataMapping/Plan/PlanProject.cs:96]: Does this need to be public?

File: AcademicTimePlanner/DataMapping/Toggl/TogglProject.cs [hunk 3]
```diff
@@ -4,7 +4,9 @@ public class TogglProject
     {
         public const long NoTogglProjectId = -1;
 
-        private List<TogglTask> _taskList;
+        private Dictionary<long, string> _taskList;
```
Comment (DuplosFidibuss) [inline AcademicTimePlanner/DataMapping/Toggl/TogglProject.cs:7]: Again here: I think `_tasks` would do it.

File: AcademicTimePlanner/DataMapping/Plan/PlanProject.cs (hunk 1 above)
Comment (Ra1nerZufall) [reply AcademicTimePlanner/DataMapping/Plan/PlanProject.cs]: no remnant of the before refactoring event.

File: AcademicTimePlanner/DataMapping/Plan/PlanProject.cs (hunk 2 above)
Comment (Ra1nerZufall) [reply AcademicTimePlanner/DataMapping/Plan/PlanProject.cs]: no remnant of the before refactoring event.

File: AcademicTimePlanner/DataMapping/Toggl/TogglProject.cs (hunk 3 above)
Comment (Ra1nerZufall) [reply AcademicTimePlanner/DataMapping/Toggl/TogglProject.cs]: changed to _task

File: AcademicTimePlanner/DataMapping/Plan/PlanProject.cs [hunk 4]
```diff
@@ -35,42 +44,99 @@ public PlanProject(long togglProjectId, string name)
...
+            _planEntries = new List<PlanEntry>();
+            _repetitionEntries = new List<PlanEntryRepetition>();
         }
 
         public PlanProject(string name)
         {
             Id = Guid.NewGuid();
             Name = name;
-            _taskList = new List<PlanTask>();
+            _task = new Dictionary<long, string>();
```
Comment (DuplosFidibuss) [inline AcademicTimePlanner/DataMapping/Plan/PlanProject.cs:56]: `_tasks`, not `_task` - there might be more than one ...
Same thing for `TogglProject`.

**Review by DuplosFidibuss — CHANGES_REQUESTED** [review]

**Review by DuplosFidibuss — APPROVED** [review]

**Review by DuplosFidibuss — APPROVED** [review]

File: AcademicTimePlanner/DataMapping/Plan/PlanProject.cs (hunk 4 above)
Comment (Ra1nerZufall) [reply AcademicTimePlanner/DataMapping/Plan/PlanProject.cs]: Now Tasks

**Pull request merged**